sdk/analytics/src/test/java/com/klaviyo/analytics/networking/requests/KlaviyoApiRequestTest.kt

```diff
@@ -129,14 +129,14 @@ internal class KlaviyoApiRequestTest : BaseApiRequestTest<KlaviyoApiRequest>() {
     }
 
     @Test
-    fun `Increments attempt counter on send`() {
-        withConnectionMock(URL(expectedFullUrl))
+    fun `Increments attempt counter on send and uses correct attempt number in header`() {
+        val connectionMock = withConnectionMock(URL(expectedFullUrl))
         val request = makeTestRequest()
 
         assertEquals(0, request.attempts)
         request.send()
         assertEquals(1, request.attempts)
-        assertEquals(request.headers["X-Klaviyo-Attempt-Count"], "1/50")
```

Comment:
The test that should have caught this was only looking at the API request object, not what was written to the url connection. Once I'd updated this test, the prior code failed. Moving `attempts++` fixes the test.

Reply:
BTW the debug logs and the internal test app also display `KlaviyoApiRequest.headers`, which must be part of why this slipped through the cracks.

Reply:
It found it while looking in API developer logs serverside. Would have also been visible in a proxy, must not have tested that way.

Reply:
Yeah I've had better luck using proxy for testing stuff related to the SDK.